messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -295,12 +281,34 @@ public synchronized Long getReaderBackLog() throws IOException {
       }
       retryCount++;
     }
-    if (lReader.getCurrentFile() != null) {
-      localStreamPendingSize += lReader.getPendingSize();
-      LOG.info("Pending Size of compressed data inside streams_local output location starting from file: " +
-              lReader.getCurrentFile() + " is " + localStreamPendingSize);
+    /**
+     * Reader can be either local stream reader or collector reader at any point of time.
+     * Based on that, the other reader's pending size to be calculated, along with current reader's
+     * pending size. Return sum of both, after adjusting local stream reader's pending size
+     * with compression factor.
+     */
+    if (reader == lReader) {
+      if (lReader.getCurrentFile() != null) { //current reader is local stream reader
+        //get local reader pending size
+        localStreamPendingSize = lReader.getPendingSize(lReader.getCurrentFile());
+        //get collector info from local stream file name and calculate collector pending size
+        String collectorPath = CollectorStreamReader.getCollectorFileName(streamName, lReader.getCurrentFile().getName());
```

Comment:
@Amareshwari  Can you please review the logic that switches from collector path to local stream path to find pending size and vice-versa?